### PR TITLE
fix(sql): fix invalid column error in group by based on function key

### PR DESCRIPTION
**Pull request by puzpuzpuz**

### CODE REVIEW

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 5 / 5 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/SqlOptimiser.java|5|5|100.00%|

File: core/src/test/java/io/questdb/griffin/KeyedAggregationTest.java [hunk 1]
```diff
@@ -1095,6 +1097,32 @@ public void testFirstLastAggregationsNotSupported() {
...
+                        "foobaz\t63\t2\n" +
+                        "bazbaz\t37\t1\n",
+                "select replace(s, 'bar', 'baz'), count(), count_distinct(s) from tab",
+                "create table tab as (select timestamp_sequence(0, 100000) ts, rnd_str('foobar','foobaz','barbaz') s from long_sequence(100))",
+                null, true, true, true
+        );
+    }
+
+    @Test
+    public void testStrFunctionKeyReverseOrder() throws Exception {
```
Comment (bluestreak01) [inline core/src/test/java/io/questdb/griffin/KeyedAggregationTest.java:1114]: this test does not fail on master, is this just an additional test or was it aimed to reproduce the problem this PR addresses?

Comment (puzpuzpuz) [reply core/src/test/java/io/questdb/griffin/KeyedAggregationTest.java:1114]: It's just a symmetrical test aimed to catch possible future regressions.

**Review by bluestreak01 — APPROVED** [review]

**Review by jerrinot — COMMENTED** [review]
the code change makes sense to me, but I don't dare to say I understand all possible side-effects. SqlOptimizer is mostly a new territory to me, and this part is particularly deep in the rabbit hole.

**Comment by puzpuzpuz** [comment]
> the code change makes sense to me, but I don't dare to say I understand all possible side-effects. SqlOptimizer is mostly a new territory to me, and this part is particularly deep in the rabbit hole. 

No problem with that, really. Possible regressions (and can't imagine any in this case - but I may be wrong) should be ideally picked up by the test suite.

**Comment by puzpuzpuz** [comment]
@bluestreak01 thanks for the review!

**Pull request merged**